ldaptor/test/test_pureldap.py

```diff
@@ -1561,3 +1561,29 @@ def test_matching_rule_assertion_with_tag_repr(self):
                                "value='rule'), type=LDAPMatchingRuleAssertion_type(value='type'), matchValue=" \
                                "LDAPMatchingRuleAssertion_matchValue(value='value'), dnAttributes=None, tag=42)"
                     self.assertEqual(repr(mra), mra_repr)
+
+    def test_ldap_bind_response_server_sasl_creds_repr(self):
+        """ ServerSaslCreds will often have binary data. A custom repr is needed because
+        it cannot be turned into a unicode string like most BEROctetString objects.
+        """
+        sasl_creds = pureldap.LDAPBindResponse_serverSaslCreds(value=b'NTLMSSP\xbe')
+        if six.PY3:
+            expected_repr = r"LDAPBindResponse_serverSaslCreds(value=b'NTLMSSP\xbe')"
+        else:
+            expected_repr = "LDAPBindResponse_serverSaslCreds(value=NTLMSSP\xbe)"
+
+        actual_repr = repr(sasl_creds)
+        self.assertEqual(actual_repr, expected_repr)
+
+    def test_ldap_bind_response_server_sasl_creds_with_tag_repr(self):
+        """ ServerSaslCreds will often have binary data. A custom repr is needed because
+        it cannot be turned into a unicode string like most BEROctetString objects.
```

Comment:
this docstring looks like a duplicate.
I don't think that using duplicate docstring add any value... it can reduce the value as it can cause confusion.


Maybe use something like this: A response with a non-standard tag will have the tag visible in the text representation.

Reply:
Good feedback! I'll update the diff :)